File: _Src/Freem/Freem.Entities.UseCases.DTO/RunningRecords/Start/StartRunningRecordRequest.cs
using Freem.Entities.Common.Relations.Collections;
using Freem.Entities.Records.Models;

namespace Freem.Entities.UseCases.DTO.RunningRecords.Start;

public sealed class StartRunningRecordRequest
{
    private readonly RelatedTagsCollection _tags = RelatedTagsCollection.Empty;
    
    public DateTimeOffset StartAt { get; }
    
    public RecordName? Name { get; init; }
    public RecordDescription? Description { get; init; }
    
    public RelatedActivitiesCollection Activities { get; }

    public RelatedTagsCollection Tags
    {
        get => _tags;
        init
        {
            ArgumentNullException.ThrowIfNull(value);

            _tags = value;
        }
    }
    
    public StartRunningRecordRequest(DateTimeOffset startAt, RelatedActivitiesCollection activities)
    {
        ArgumentNullException.ThrowIfNull(activities);
        
        StartAt = startAt;
        Activities = activities;
    }
}

File: _Src/Freem/Freem.Web/src/clients/models/records/running/StartRunningRecordRequest.ts
export class StartRunningRecordRequest {
  public readonly startAt: string;
  public readonly activities: string[];

  public readonly name?: string;
  public readonly description?: string;
  public readonly tags?: string[];

  constructor(
    startAt: string,
    activities: string[],
    name?: string,
    description?: string,
    tags?: string[]
  ) {
    this.startAt = startAt;
    this.activities = activities;

    this.name = name;
    this.description = description;
    this.tags = tags;
  }
}